packages/core/admin/admin/src/content-manager/pages/EditView/components/DocumentActions.tsx

```diff
@@ -554,7 +608,13 @@ const UnpublishAction: DocumentActionComponent = ({ activeTab, id, model, collec
     }),
     icon: <StyledCrossCircle />,
     onClick: async () => {
-      if (!id) {
+      if ((!id && collectionType !== SINGLE_TYPES) || isDocumentModified) {
+        if (!id) {
+          // This should never, ever, happen.
+          throw new Error(
+            "You're trying to unpublish a document without an id, this is likely a bug with Strapi. Please open an issue."
+          );
```

Comment:
This is just going to blow up right? Should this be an alert with a translated message?

Reply:
IMO it should never happen, we could console error, but the user won't get any feedback immediately, i'm not sure about it being an alert... but maybe it should generic error notification...

Reply:
Up to you it's a small detail so cool to leave it as is as well